### PR TITLE
Add layer reuse metrics

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This adds a new metric `crio_image_layer_reuse` which collects the
amount of reu-used already locally available image digests.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
- Added `crio_image_layer_reuse` metric which counts layer reuses during image pull
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Oh my, green tests!

@haircommander @mrunalp PTAL

File: server/image_pull.go [hunk 1]
```diff
@@ -272,7 +279,19 @@ func tryIncrementImagePullFailureMetric(ctx context.Context, img string, err err
...
+
+	counter, err := metrics.CRIOImageLayerReuse.GetMetricWithLabelValues(layer)
+	if err != nil {
+		log.Warnf(ctx, "Unable to write image layer reuse metrics: %v", err)
+	} else {
+		counter.Inc()
+	}
+}
+
+func decodeDockerAuth(s string) (user, password string, err error) {
```
Comment (haircommander) [inline server/image_pull.go:294]: why'd this change to err?

Comment (saschagrunert) [reply server/image_pull.go:294]: That’s the rest of the merge conflict resolution. I’ll revert it.

**Comment by TomSweeneyRedHat** [comment]
LGTM
and me thinks you spoke too soon on the green tests @saschagrunert :^(

**Pull request merged**